addons/web/static/tests/views/list_view_tests.js

```diff
@@ -15887,4 +15989,323 @@ QUnit.module("Views", (hooks) => {
         await click(target, ".o_field_many2one_selection .o-autocomplete--input");
         assert.verifySteps(["name_search"]);
     });
+
+
+
+    // Properties fields
+
+    QUnit.test("properties fields columns are replaced by the columns of their definitions", async function (assert) {
+        assert.expect(3);
+
+        await makeView({
+            type: "list",
+            resModel: "foo",
+            serverData,
+            arch: `
+                <tree>
+                    <field name="m2o" invisible="1"/>
+                    <field name="second_m2o_bar" invisible="1"/>
+                    <field name="some_properties_field"/>
+                    <field name="some_other_properties_field"/>
+                </tree>
+            `,
+        });
+
+        const propertiesColumns = [...target.querySelectorAll("th[data-name^='some_properties_field'],th[data-name^='some_other_properties_field']")];
+        const allColumns = [...target.querySelectorAll("table.o_list_table thead th[data-name]")];
+
+        assert.ok(propertiesColumns, "There should be some columns with a name starting in the form 'properties.xxx'");
+        assert.deepEqual(propertiesColumns.length, allColumns.length, 'All properties fields should have been replaced with their definitions');
+        assert.deepEqual(new Set(propertiesColumns), new Set(allColumns));
+    });
+
+    QUnit.test("properties with no value have columns", async function (assert) {
+        assert.expect(1);
+
+        // only display the empty property of the first record
+        await makeView({
+            type: "list",
+            resModel: "foo",
+            serverData,
+            arch: `
+                <tree limit="1">
+                    <field name="second_m2o_bar" invisible="1"/>
+                    <field name="some_other_properties_field"/>
+                </tree>
+            `,
+        });
+
+        const propertiesColumns = [...target.querySelectorAll("th[data-name^='some_other_properties_field']")];
+
+        assert.deepEqual(propertiesColumns.length, 2, 'All properties defined in the displayed records should have a column even when no record has a value for it');
```

Comment:
can you check the column names ? (and the order as well)

Reply:
Yep, also added a proper check on the technical name and the values since I'd missed these too.